core/templates/dev/head/pages/collection_editor/CollectionEditorNavbarDirective.js

```diff
@@ -68,11 +78,15 @@ oppia.directive('collectionEditorNavbar', [function() {
           // action since it is not reversible.
           CollectionRightsBackendApiService.setCollectionPublic(
             $scope.collectionId, $scope.collection.getVersion()).then(
-            function() {
-              // TODO(bhenning): There should be a scope-level rights object
-              // used, instead. The rights object should be loaded with the
-              // collection.
-              $scope.isPrivate = false;
+            function(data) {
+              if (!data.is_private) {
+                $scope.isPrivate = data.is_private;
+                $rootScope.$broadcast(EVENT_COLLECTION_STATUS_CHANGE);
```

Comment:
This is a bit ick, yeah. I don't think we should do it. At the very least the $broadcast should come from the service; having it come from the navbar is a bit random.

That said, probably the 'correct' way is to either (a) have a rights dict exposed by the service that the components can bind to, so that they refer directly to collectionRights.status, or (b) have the settings page card query a method in the service, something like CollectionRightsDataService.isPublic().

Could you maybe ping @BenHenning and get his thoughts on this?